OpenRA.Mods.Common/Activities/Move/AttackMoveActivity.cs

```diff
@@ -20,29 +20,71 @@ public class AttackMoveActivity : Activity
 	{
 		const int ScanInterval = 7;
 
+		readonly CPos destination;
+		readonly int nearEnough;
 		Activity inner;
-		int scanTicks;
+		Activity attack;
 		AutoTarget autoTarget;
+		bool moving;
 
-		public AttackMoveActivity(Actor self, Activity inner)
```

Comment:
In one of my prototypes I replaced this with a Func that returned a new `Activity` instead. That let's you keep the flexibility.

Reply:
I'm completely fine with dropping this provided we don't have a specific usecase that needs it now.  It is Attack*Move*, after all.

Reply:
Thanks for the tip @obrakmann ! Like I said, I'm not very experienced in C# so I didn't now this was possible. There is indeed one usecase for this and that is Guard orders. I have now folded GuardActivity back into AttackMoveActivity and used a delegate to handle this. This reduces the amount of duplicate code.